compiler/rustc_target/src/target_features.rs

```diff
@@ -582,6 +582,12 @@ pub fn all_rust_features() -> impl Iterator<Item = (&'static str, Stability)> {
 const X86_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] =
     &[(128, "sse"), (256, "avx"), (512, "avx512f")];
 const AARCH64_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(128, "neon")];
+const POWERPC_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(128, "altivec")];
+const RISCV_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] =
+    &[(32, "zve32x"), (64, "zve64x"), (128, "zvl128b")];
```

Comment:
These target features are not included in RISCV_FEATURES.

Reply:
Good point - what do you think is the best option here? Warn unconditionally for now, and once the relevant features are added, change the list?

Reply:
So how to people do vector stuff on Risc-V currently?

Reply:
> So how do people do vector stuff on Risc-V currently?

See https://github.com/rust-lang/rust/issues/131800#issuecomment-2418361508.

In any case, as with SPARC, the only vector types currently available for this target are core::simd and custom repr(simd) types, which are unstable.

(Also, I don't think we can add corresponding vector types to core::arch right away because the vector types enabled by V (and related) extension are scalable vectors which cannot yet exist as Rust types, and the P extension is still draft.)